src/material/schematics/ng-update/migrations/theming-api-v12/theming-api-migration.ts

```diff
@@ -37,7 +37,9 @@ export class ThemingApiMigration extends DevkitMigration<null> {
         this.fileSystem
           .edit(stylesheet.filePath)
           .remove(0, stylesheet.content.length)
-          .insertLeft(0, migratedContent);
+          // Strip the BOM to avoid issues with the Sass compiler. See:
+          // https://github.com/angular/components/issues/24227#issuecomment-1200934258
+          .insertLeft(0, stripBom(migratedContent));
```

Comment:
I believe the whole issue surfaced because (assuming `X` is the BOM) has been shifted in the theming API migration from `idx: 0` to somewhere in the middle of the file.

Stripping the BOM afterwards like this won't solve this I think -- because the RegExp only strips the BOM at `idx: 0`. The fix would be to strip the BOM before starting the migration (i.e. first step of the input transformation).

I think, similarly to TS, we could strip the BOM always as part of the `visitStylesheet`/`ResolvedResource`

Reply:
Ah, you're right, I didn't notice the `^` in the regex. I wanted to do it as the last step of the migration so that it doesn't affect files that we aren't migrating.

Regarding moving it to `visitStylesheet`, I was looking for something similar, but I'm not sure that doing it in `Migration.visitStylesheet` would help since sub classes usually override the method. We could potentially do it in `UpdateProject.migrate` before the content is passed to the individual migrations.

Reply:
For TS we also do it before any migrations run, so that would be equivalent. I think we should strip it off as part of `ResolvedResource`

Reply:
Done. I only kept it to stylesheets for now since we haven't seen similar issues in other file types.

Reply:
sgtm